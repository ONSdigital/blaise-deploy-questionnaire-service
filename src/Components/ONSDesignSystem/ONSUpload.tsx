import React, {ChangeEvent, Component} from "react";

interface Props{
  label: string,
  description: string,
  id?: string
  fileName: string,
  fileID: string,
  accept: string,
  onChange? : (e: ChangeEvent<HTMLInputElement>, ...args: any[]) => void
}

export class ONSUpload extends Component <Props >{

    value = "";
    constructor(props : Props) {
        super(props);
        this.state = {value: ""};
    }

    handleChange = (e: ChangeEvent<HTMLInputElement>) => {
        if(this.props.onChange !== undefined){
            this.props.onChange(e);
        }
        this.value = e.target.value;
    };

    render() {
      return (
        <div className="field">
            <p className="field">
                <label className="label" htmlFor={this.props.fileID}>{this.props.label}
                <br/>
                  <span className="label__description">{this.props.description}</span>
                </label>
                <input style={{position: "static"}} type="file" id={this.props.fileID} className="input input--text input-type__input input--upload" name={this.props.fileName} accept={this.props.accept} onChange={(e) => this.handleChange(e)}/>
            </p>
        </div>
      );
  }
}
