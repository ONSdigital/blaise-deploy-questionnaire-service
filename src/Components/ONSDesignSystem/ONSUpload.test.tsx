import React from "react";
import Enzyme, {mount, shallow} from "enzyme";
import {ONSUpload} from "./ONSUpload";
import Adapter from "enzyme-adapter-react-16";

describe("ONS Upload Test", () => {
    Enzyme.configure({ adapter: new Adapter() });

    const Props = {

    }

    const changeProps = {
        label: "Upload",
        description: "This is the upload",
        fileName: "file.csv",
        fileID: 'file',
        accept: 'yes?',
        onChange: jest.fn()
    }

    const undefinedChangeProps = {
        label: "Upload",
        description: "This is the upload",
        fileName: "file.csv",
        fileID: 'file',
        accept: 'yes?',
        onChange: undefined
    }

    function wrapper(render: any, props: any) {
        return render(
            <ONSUpload
                label={props.label}
                description={props.description}
                fileName={props.fileName}
                fileID={props.fileID}
                accept={props.accept}
                onChange={props.onChange}>
            </ONSUpload>
            
        )
    }

    it("matches Snapshot", () => {
        expect(wrapper(shallow, Props)).toMatchSnapshot()
    });

    it("should render correctly", () => expect(wrapper(shallow, Props).exists()).toEqual(true));

    it("should handle a change", () => {
        //defined onchange
        wrapper(mount, changeProps).find('input').simulate('change')
        expect(changeProps.onChange).toHaveBeenCalled()        
        
        //undefined onchange
        wrapper(mount, undefinedChangeProps).find('input').simulate('change')
        expect(undefinedChangeProps.onChange).toBeUndefined()        

    })

})