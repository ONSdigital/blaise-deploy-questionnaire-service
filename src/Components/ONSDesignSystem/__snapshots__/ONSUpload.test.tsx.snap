// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ONS Upload Test matches Snapshot 1`] = `ShallowWrapper {}`;
